helix-term/src/commands/typed.rs

```diff
@@ -982,6 +982,41 @@ fn reload(
     })
 }
 
+fn lsp_restart(
+    cx: &mut compositor::Context,
+    _args: &[Cow<str>],
+    event: PromptEvent,
+) -> anyhow::Result<()> {
+    if event != PromptEvent::Validate {
+        return Ok(());
+    }
+
+    let (_view, doc) = current!(cx.editor);
+    let config = doc
+        .language_config()
+        .ok_or_else(|| anyhow!("LSP not defined for current document"))?;
+
+    let scope = config.scope.clone();
+    cx.editor.language_servers.restart(config)?;
+
+    // This collect is needed because refresh_language_server would need to re-borrow editor.
+    #[allow(clippy::needless_collect)]
+    let to_refresh: Vec<DocumentId> = cx
+        .editor
+        .documents()
+        .filter_map(|doc| match doc.language_config() {
+            Some(config) if config.scope.eq(&scope) => Some(doc.id()),
+            _ => None,
+        })
+        .collect();
+
+    to_refresh.into_iter().for_each(|id| {
+        cx.editor.refresh_language_server(id);
+    });
```

Comment:
```suggestion
    let document_ids_to_refresh: Vec<DocumentId> = cx
        .editor
        .documents()
        .filter_map(|doc| match doc.language_config() {
            Some(config) if config.scope.eq(&scope) => Some(doc.id()),
            _ => None,
        })
        .collect();

    for document_id in document_ids_to_refresh {
        cx.editor.refresh_language_server(document_id);
    }
```

I think this would be a bit more idiomatic plus it fixes the clippy lint.